app/services/upload_sanitizer.rb

```diff
@@ -0,0 +1,19 @@
+# Formats uploaded files to UTF-8 encoding and strips unexpected BOM characters.
+# Takes an open File object as input
+class UploadSanitizer
+  def initialize(upload)
+    @data = upload
+  end
+
+  def call
+    @data.force_encoding('UTF-8')
+    strip_bom_character
+    @data
+  end
+
+  private
+
+  def strip_bom_character
+    @data.gsub!("\xEF\xBB\xBF".force_encoding("UTF-8"), '')
```

Comment:
IMO it'd be better to use its non-bang version to avoid the mutation which will make the line 11 unnecessary. This would return the stripped string already.

Reply:
Done @sauloperez